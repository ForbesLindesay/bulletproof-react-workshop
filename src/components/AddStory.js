// @flow

import React from 'react';

type Props = {|
  +body: string,
  +onSubmit: (e: SyntheticInputEvent) => mixed,
  +onChangeBody: (e: SyntheticInputEvent) => mixed,
|};

export default function AddStory(props: Props) {
  return (
    <form onSubmit={props.onSubmit}>
      <input
        onChange={props.onChangeBody}
        value={props.body}
        data-test-id="NewStoryBody"
      />
      <button type="submit" data-test-id="AddStoryButton">Add Story</button>
    </form>
  );
}
