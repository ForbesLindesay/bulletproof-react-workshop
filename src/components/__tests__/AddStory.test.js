// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import AddStory from '../AddStory';

test('AddStory', () => {
  expect(
    renderer
      .create(
        <AddStory
          body="A great story!"
          onChangeBody={() => {}}
          onSubmit={() => {}}
        />,
      )
      .toJSON(),
  ).toMatchSnapshot();
});
