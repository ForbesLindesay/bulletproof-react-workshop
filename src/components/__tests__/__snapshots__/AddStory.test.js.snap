// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddStory 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    data-test-id="NewStoryBody"
    onChange={[Function]}
    value="A great story!"
  />
  <button
    data-test-id="AddStoryButton"
    type="submit"
  >
    Add Story
  </button>
</form>
`;
